broker/providers/__init__.py

```diff
@@ -65,18 +65,22 @@ def __new__(cls, name, bases, attrs):
             for attr, obj in attrs.items():
                 if attr == "provider_help":
                     # register the help options based on the function arguments
-                    for name, param in inspect.signature(obj).parameters.items():
-                        if name not in ("self", "kwargs"):
-                            # {name: (cls, is_flag)}
-                            PROVIDER_HELP[name] = (
+                    for _name, param in inspect.signature(obj).parameters.items():
+                        if _name not in ("self", "kwargs"):
+                            # {_name: (cls, is_flag)}
+                            PROVIDER_HELP[_name] = (
                                 new_cls,
                                 isinstance(param.default, bool),
                             )
-                            logger.debug(f"Registered help option {name} for provider {name}")
+                            logger.debug(f"Registered help option {_name} for provider {_name}")
                 elif hasattr(obj, "_as_action"):
                     for action in obj._as_action:
                         PROVIDER_ACTIONS[action] = (new_cls, attr)
                         logger.debug(f"Registered action {action} for provider {name}")
+            # register provider settings validators
+            new_validators = [v for v in attrs.get("_validators") if v not in settings.validators]
+            logger.debug(f"Adding new validators: {[v.names[0] for v in new_validators]}")
+            settings.validators.extend(new_validators)
```

Comment:
We could theoretically simplify the loading logic further with this.
```suggestion
            if "_validators" in attrs:
                settings.validators.extend(attrs["_validators"])
```